vortex-array/src/array/bool/mod.rs

```diff
@@ -37,14 +37,19 @@ impl BoolArray {
 
 impl BoolArray {
     pub fn try_new(buffer: BooleanBuffer, validity: Validity) -> VortexResult<Self> {
+        let buffer_len = buffer.len();
+        let buffer_offset = buffer.offset();
+
+        let inner = buffer.into_inner().bit_slice(buffer_offset, buffer_len);
```

Comment:
So I think this only partially fixes the bug. The problem is that the BoolArray itself doesn't hold an offset, meaning the first bit is always considered to be byte aligned.

Maybe in another PR can fix for real by adding a bit_offset (from 0..8) field to BoolMetadata. The SliceFn compute function and maybe others must also respect it.

Reply:
`bit_slice` actually copies the buffer if the offset isn't byte aligned so this operation isn't as cheap as it used to be. Whether that's a reasonable tradeoff is up to you.
I do agree with the second point, I'm obviously not familiar with every part of the codebase here but taking the offset logic management into vortex will probably be valuable.

Reply:
Yeah, the general idea is that construction and SliceFn should be constant time and not re-allocate. Which is why lots of arrays hold offsets.

Reminds me, we should have some logic to slice out those offsets during serialization!